### PR TITLE
CVL-9: Switch to generated types for the licence API. 

### PR DESCRIPTION
**Pull request by timharrison-moj**
Also: 

* Tidy prisoner image routes and profile page. 
* Allow ingress from all IP addresses in DEV (experimental).
* Add prison-api to docker-compose file for local running.

### CODE REVIEW

File: helm_deploy/values-dev.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ generic-service:
   replicaCount: 2
 
   ingress:
+    enable_allow_list: false
     host: create-and-vary-a-licence-dev.hmpps.service.justice.gov.uk
```
Comment (timharrison-moj) [inline helm_deploy/values-dev.yaml:9]: Experimental. I don't think this is part of the `generic-service` helm chart but trying anyway. If this does not work, I'll comment the allow list entries and add one, over-arching allow-all rule.

File: server/@types/licenceApiImport/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      }
+      /** Forbidden, requires an appropriate role */
+      403: {
+        content: {
+          'application/json': components['schemas']['TestData'][]
+        }
+      }
+    }
+  }
+}
```
Comment (timharrison-moj) [inline server/@types/licenceApiImport/index.d.ts:42]: Generated file.

File: server/routes/otherRoutes.ts [hunk 3]
```diff
@@ -44,15 +44,19 @@ export default class OtherRoutes {
...
+        .then(data => {
+          res.type('image/jpeg')
+          data.pipe(res)
+        })
+        .catch(error => {
+          res.sendFile(placeHolder)
+        })
+    } else {
+      res.sendFile(placeHolder)
+    }
```
Comment (timharrison-moj) [inline server/routes/otherRoutes.ts:60]: Caters for missing NomsId too - probation cases may never have been into prison.

File: helm_deploy/values-dev.yaml [hunk 4]
```diff
@@ -15,6 +15,8 @@ generic-service:
     PRISON_API_URL: "https://api-dev.prison.service.justice.gov.uk"
     LICENCE_API_URL: "https://create-and-vary-a-licence-api-dev.hmpps.service.justice.gov.uk"
 
+  allowlist: null
```
Comment (yasinmustafa) [inline helm_deploy/values-dev.yaml:18]: Does this allow access from anywhere?

File: server/routes/otherRoutes.ts (hunk 3 above)
Comment (yasinmustafa) [reply server/routes/otherRoutes.ts:60]: I must remember to copy this!

**Review by yasinmustafa — APPROVED** [review]
LGTM - with a question

File: helm_deploy/values-dev.yaml (hunk 4 above)
Comment (timharrison-moj) [reply helm_deploy/values-dev.yaml:18]: Yes.

**Pull request merged**